scalafmt-macros/shared/src/main/scala/org/scalafmt/config/DialectMacro.scala

```diff
@@ -0,0 +1,33 @@
+package org.scalafmt.config
+
+import scala.reflect.macros.blackbox
+import scala.language.experimental.macros
+
+import scala.meta.Dialect
+
+// Builds a map between string (the scalafmt method name)
+// and dialect method application
+object DialectMacro {
+  def dialectMap: Map[String, ((Dialect, Any) => Dialect)] = macro dialectMap_impl
+
+  def dialectMap_impl(c: blackbox.Context): c.Expr[Map[String, ((Dialect, Any) => Dialect)]] = {
+    import c.universe._
+    val methods = typeOf[Dialect].members.flatMap {
+      case v: MethodSymbol =>
+        v.paramLists match {
+          case (param :: Nil) :: Nil => // single parameter
+            val methodName = v.name.decodedName.toString
+            if (methodName.startsWith("with")) {
+              val tpe = param.typeSignature
+              Some(q"$methodName -> ((dialect: scala.meta.Dialect, v: Any) => dialect.${TermName(methodName)}(v.asInstanceOf[$tpe]))")
```

Comment:
regarding TermName invocation here: does that mean that `v.name` can't be used instead? i assumed it would be a name, and likely not a type name....

Reply:
would like to hear your thoughts on this question

Reply:
v.name.decodedName is of the `Name` type, but the `quasiquote` expects the `TermName` type. I have not found any easier way to convert between those two (however, I am not an expert on scala 2 macros unfortunately, so I am not 100% sure)

Reply:
but `v.name` (without `.decodedName`) is of type `NameType` which for any `TermSymbol` (including `MethodSymbol`) is `TermName`.

what's the difference between `.name` and `.name.decodedName`? is it possible that `v.name` is actually the right thing to call?

Reply:
Ah, I see just using .name works now, I had some problems there doing it with the previous method (iterating through vals instead of methods with "with"), but now I've changed to just v.name